Assets/Tests/InputSystem/CoreTests_Controls.cs

```diff
@@ -488,29 +487,47 @@ public void Controls_ValueCachingWorksAcrossEntireDeviceMemoryRange()
         InputSystem.actions?.Disable();
 #endif
 
-        var keyboard = InputSystem.AddDevice<Keyboard>();
+        // Value caching is supported for all controls, but with changes to button press detection, ButtonControls will
```

Comment:
Good catch, thanks for updating this.

Reply:
However, this makes me wonder, will this evaluation work for detecting "button behavior presses" for controls that are not buttons? E.g. transition of axis under the button behavior (magnitude based) within a sub-frame? Basically this is not unique to buttons in any way, it's all about avoiding aliasing in the sampling process. Not saying these changes not already carry a lot of value, but trying to understand what problems still exist. Hence I wonder if we should setup an additional test case similar to the one done for wasPresseed/wasReleased this frame w.r.t. e.g. axis, or maybe you already can say what we guarantee for that use case? (We could just say, use actions for something like that)

Reply:
I believe what you mention will be detected just fine, eg. gamepad has a ButtonControl for "left stick up"?